### ISSUE TITLE
PVC requests don't seem to fire to heketi

### ISSUE THREAD

**Issue by mjschmidt** [gluster/gluster-kubernetes#450]
Sometimes it seems like pvcs aren't getting to heketi and as a result the pvcs sit in a forever pending state.

How can I go about debugging this?

**Comment by jarrpa** [comment]
You can check several places:
* Look for error events in `kubectl describe pvc <problem_pvc>`
* Verify the correct parameters in `kubectl get sc <gluster_sc> -o yaml`
* Look for errors in `kubectl logs <heketi_pod>`

**Comment by mjschmidt** [comment]
Okay so for the longest time my pvc was sitting forever in pending state. Finally this morning I came back and found this error:
Warning ProvisioningFailed 1342x over 5h From persistent volume controller 

Message: Failed to provision volume with storage class glusterfs-storage create volume error: error creating volume Post http://<IP_HERE>:8080/volumes: dial tcp <ip_here>:8080: getsockopt: no route to host

**Comment by mjschmidt** [comment]
gluster service for my pod:
labels:
  gluster.kubernetes.io/provisioned-for-pvc: journaldisk-bookie-0
name
namespace
resourceversion
selfLink: /api/v1/namespaces/default/glusterfs-dynamic-journaldisk-bookie-0
uid: <numberhere>
spec:
  clusterIP : 10:111:124:22
  ports:
  -  port: 1
     protocal: TCP
     targetPort: 1
  sessionAffinity: None
  type: ClusterIP
status:
  loadbalencer: {}

**Comment by jarrpa** [comment]
I said `sc` as in `storageclass`.

**Comment by mjschmidt** [comment]
Sorry
Name glusterfs-storage
IsDefaultClass: No
Annotations: kubectl.kubernetes.io/last-applied-configuration={apiversion:storage.k8s.io/v1beta1, kind: sotrageclass, metadata: { annotations: {}, name: glusterfs-storage, namespace "", parameters: {resturl: http://10.244.7.2:8080}, provisioner: kubernetes.io/glusterfs

Provisioner: kubernetes/glusterfs
Parameters: resturl=http://10.244.7.2:8080
Events: <none>

**Comment by jarrpa** [comment]
See if you get a response to `curl http://10.244.7.2:8080/hello` from one of your master nodes. If not you have some underlying networking problems.

**Comment by mjschmidt** [comment]
Yeah that worked, but after some googling I saw that there were issues using kubernetes 1.8.2-1.8.0 with pvc stuff. So I am working on going up to 1.9.3 right now and seeing if that solves things before I continue debugging

**Comment by mjschmidt** [comment]
I figured out the issue which is captured here: https://github.com/gluster/gluster-kubernetes/issues/434
